src/server/jobQueuesUI.js

```diff
@@ -20,7 +21,11 @@ const toureiroApp = toureiro({
 app.use(
   '/job-queues',
   (req, res, next) => {
-    if (!req.user || !userCan(req.user, 'monitorJobQueues')) {
+    if (!req.user) {
+      const redirectUrl = `${req.protocol}://${req.get('host')}${req.originalUrl}`
+      res.redirect(loginURL({redirect: redirectUrl}))
+    }
+    if (!userCan(req.user, 'monitorJobQueues')) {
       throw new LGNotAuthorizedError()
```

Comment:
Also need to pass the error in `next()` instead of throwing (I think this was preexisting, but let's fix it while we're in here):

```js
return next(new LGNotAuthorizedError())
```